benchmarks/high_volume_writes_test.go

```diff
@@ -50,69 +57,55 @@ var _ = Describe("Scenario: High Volume Writes", func() {
-		defaultRange := scenarioCfg.Samples.Range
-
-		//
-		// Collect measurements for the distributor
-		//
-		job := benchCfg.Metrics.DistributorJob()
-
-		// Record Reads QPS
-		qps, err := metricsClient.RequestWritesQPS(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read QPS for all distributor push with status code 2xx")
-		b.RecordValue("All distributor 2xx push QPS", qps)
-
-		// Record p99 loki_request_duration_seconds_bucket
-		p99, err := metricsClient.RequestDurationOkPushP99(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read p99 for all distributor push requests with status code 2xx")
-		b.RecordValue("All distributor 2xx push p99", p99)
-
-		// Record p50 loki_request_duration_seconds_bucket
-		p50, err := metricsClient.RequestDurationOkPushP50(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read p50 for all distributor push requests with status code 2xx")
-		b.RecordValue("All distributor 2xx push p50", p50)
-
-		// Record avg from loki_request_duration_seconds_sum / loki_request_duration_seconds_count
-		avg, err := metricsClient.RequestDurationOkPushAvg(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read average for all distributor push requests with status code 2xx")
-		b.RecordValue("All distributor 2xx push avg", avg)
-
-		//
-		// Collect measurements for the ingester
-		//
-		job = benchCfg.Metrics.IngesterJob()
-
-		// Record Writes QPS
-		qps, err = metricsClient.RequestWritesGrpcQPS(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read QPS for all ingester GRPC push with status code 2xx")
-		b.RecordValue("All ingester successful GRPC push QPS", qps)
-
-		// Record BoltDB Shipper Writes QPS
-		qps, _ = metricsClient.RequestBoltDBShipperWritesQPS(job, defaultRange)
-		b.RecordValue("All boltdb shipper successful writes QPS", qps)
-
-		// Record p99 loki_request_duration_seconds_bucket
-		p99, err = metricsClient.RequestDurationOkGrpcPushP99(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read p99 for all ingester GRPC push with status code 2xx")
-		b.RecordValue("All ingester successful GRPC push p99", p99)
-
-		// Record p50 loki_request_duration_seconds_bucket
-		p50, err = metricsClient.RequestDurationOkGrpcPushP50(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read p50 for all ingester GRPC push with status code 2xx")
-		b.RecordValue("All ingester successful GRPC push p50", p50)
-
-		// Record avg from loki_request_duration_seconds_sum / loki_request_duration_seconds_count
-		avg, err = metricsClient.RequestDurationOkGrpcPushAvg(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read average for all ingester GRPC push with status code 2xx")
-		b.RecordValue("All ingester successful GRPC push avg", avg)
-
-		mu.Lock()
-		defer mu.Unlock()
-		totalSamples -= 1
-
-	}, benchCfg.Scenarios.HighVolumeWrites.Samples.Total)
+	for _, configuration := range benchCfg.Scenarios.HighVolumeWrites.Configurations {
```

Comment:
This for-range-loop looks like registering one `Measure(...)` block per scenario configuration. How does it ensure that they are not run interleaved? FWIW, `Measure()` registers only the `func(b Benchmarker){...}` to ginkgo to be executed `c.Sample.Total` times.

Reply:
The idea here is to have new Measure registration for every different test Configuration so that we get separate graphs for every different configuration. The loop https://github.com/observatorium/loki-benchmarks/blob/41f950088a2939159b4db48f78a5990bb10cdf3c/benchmarks/high_volume_writes_test.go#L67 ensures that the measure function is re-registered on every new configuration ... the step  `totalSamples -= 1` for each of the c.Samples.Total is aligned with the `if totalSamples == 0` code under `BeforeEach` code so that  when one configuration ends, `totalSamples` is updated to next `Samples.Total` and logger reploy with the new stress level.